src/poetry/core/version/markers.py

```diff
@@ -443,34 +430,20 @@ def of(cls, *markers: BaseMarker) -> BaseMarker:
 
         return MultiMarker(*new_markers)
 
-    @property
-    def markers(self) -> list[BaseMarker]:
-        return self._markers
-
     def intersect(self, other: BaseMarker) -> BaseMarker:
-        if other.is_any():
-            return self
-
-        if other.is_empty():
-            return other
+        multi = MultiMarker(self, other)
+        return dnf(multi)
 
+    def union(self, other: BaseMarker) -> BaseMarker:
         if isinstance(other, MarkerUnion):
-            return other.intersect(self)
-
-        new_markers = self._markers + [other]
-
-        multi = MultiMarker.of(*new_markers)
+            return other.union(self)
```

Comment:
Can we drop this? It seems it just changes nothing but the order (and IMO without it, it's even better).

Reply:
doing so results in union being not-quite-symmetrical, in a way that the unit tests notice.  Similar in the other one.

Perhaps bowing to the unit tests is allowing the tail to wag the dog though